### ISSUE TITLE
Song editor: shorten notes to create some space between notes

### ISSUE THREAD

**Issue by achimmihca** [UltraStar-Deluxe/Play#174]
In some songs I have the problem with notes directly adjacent to one another without space between. This is sometimes wrong because it is missing the singers breathing between notes for example.

A new feature for the song editor could be an editing option to shorten notes, such that a minimum amount of space is between two notes.
If notes are selected, then only the selected notes should be considered in this operation. Otherwise, all notes should be considered in the operation.
The operation should consider notes next to each other, ignoring difference in pitch.

The minimum amount of space could be defined in beats or seconds. In seconds would be BPM independent.

**Comment by bohning** [comment]
I think this is would be a valuable feature. It reminds me of the USDX CMD version in two regards:

1. CMD offered the possibility to adjust the BPM while recalculating note length accordingly (very great feature for wrong BPMs). I used this feature to adjust adjacent notes (due to rounding, I only had to recalculate to a lower value and then go back to the original value, if I remember correctly, then adjacent notes were separated by one beat, but of course this was a side effect and not the intended use of the feature).

2. CMD (if I remember correctly, couldn’t find it in the change log) offered some sort of "quality analysis" of songs that could be displayed in-game and also exported (to *.xls?). This considered features such as BPM, percentage of adjacent notes, existence of golden notes, line break timing errors, etc. which gave an overall score, I think. If that’s somewhat interesting, I can try to research some more.

By the way, here’s the full changelog of the CMD, maybe as an inspiration of good next features: https://github.com/UltraStar-Deluxe/USDX/issues/361.
